plugins/orchestrator-backend/src/service/DataIndexService.ts

```diff
@@ -56,24 +71,110 @@ export class DataIndexService {
             }
         }
       `;
-  }
 
-  public static async getWorkflowDefinitions(): Promise<WorkflowDefinition[]> {
-    if (!this.inited) {
-      throw ErrorBuilder.GET_SWF_BACKEND_NOT_INITED();
-    }
-    const QUERY = this.getDeployedSwfsQuery();
-    this.backendExecCtx.logger.info(
+    this.logger.info(
       `getWorkflowDefinitions() called:  ${this.backendExecCtx.dataIndexUrl}`,
     );
-    const result = await this.backendExecCtx.client.query(QUERY, {});
+    const result = await this.client.query(QUERY, {});
 
     if (result.error) {
-      this.backendExecCtx.logger.error(
+      this.logger.error(
         `Error fetching data index swf results ${result.error}`,
       );
       throw result.error;
     }
+
     return result.data.ProcessDefinitions;
   }
+
+  public async fetchProcessInstances(): Promise<ProcessInstance[] | undefined> {
+    const graphQlQuery = `{ ProcessInstances ( orderBy: { start: ASC }, where: {processId: {isNull: false} } ) { id, processName, processId, state, start, lastUpdate, end, nodes { id }, variables, parentProcessInstance {id, processName, businessKey} } }`;
```

Comment:
Please also include the following change related to `businessKey` in the queries.
These changes have been merged yesterday.

<img width="1532" alt="image" src="https://github.com/caponetto/backstage-plugins/assets/638737/921c1ca4-b7d0-4b91-977c-080a45e9d8fa">